starky/src/vanishing_poly.rs

```diff
@@ -0,0 +1,66 @@
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::hash::hash_types::RichField;
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+use plonky2::plonk::config::GenericConfig;
+
+use crate::config::StarkConfig;
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::permutation::{
+    eval_permutation_checks, eval_permutation_checks_recursively, PermutationCheckData,
+    PermutationCheckDataTarget,
+};
+use crate::stark::Stark;
+use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
+
+pub(crate) fn eval_vanishing_poly<F, FE, C, S, const D: usize, const D2: usize>(
```

Comment:
Let's have this supported packed fields - it's a small change, cb749ed7554e9f1cd249b1f8ce207e4e49a2709c